code/game/objects/items/kirbyplants.dm

```diff
@@ -42,6 +37,11 @@
 		if(do_after(user,3 SECONDS,target=src))
 			qdel(I)
 			dead = FALSE
+			var/obj/item/kirbyplants/ref_plant = /obj/item/kirbyplants
+			name = initial(ref_plant.name)
+			desc = initial(ref_plant.desc)
```

Comment:
for future ref, this should ALWAYS be handled in `update_desc()` (which gets called when you call `update_apperance()`)